datafusion/common/src/scalar/mod.rs

```diff
@@ -854,6 +854,140 @@ pub fn get_dict_value<K: ArrowDictionaryKeyType>(
     Ok((dict_array.values(), dict_array.key(index)))
 }
 
+/// Cache for dictionary key arrays to avoid repeated allocations
```

Comment:
I think it is important to also point out that the cached keys are always zero

SO like "cache for dictionary keys for single valued dictionary arrays" or something